scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/license/LicenseSchedulingPolicy.java

```diff
@@ -53,110 +54,97 @@
 
     private static final String REQUIRED_LICENSES = "REQUIRED_LICENSES";
 
-    // A map of fixed size lists. A map entry per software license and a list per map entry to handle jobs using this software license.
-    private Map<String, LinkedBlockingQueue<JobDescriptorImpl>> eligibleJobsDescriptorsLicenses = null;
-
-    // A map of fixed size lists. A map entry per software license and a list per map entry to handle tasks using this software license.
-    private Map<String, LinkedBlockingQueue<EligibleTaskDescriptor>> eligibleTasksDescriptorsLicenses = null;
-
-    private Map<String, Integer> remainingTokens = null;
-
-    // Will be initialize with the license properties file which includes the maximum licenses numbers per software
+    // Will be initialize with the license properties file which includes the number of tokens per license
     private Properties properties = null;
 
+    // For DB persistence
+    private LicenseSynchronization licenseSynchronization = null;
+
     private void initialize() {
 
-        if (remainingTokens == null) {
+        if (properties == null) {
 
             logger.debug("Retrieve properties from the license properties file");
             properties = LicenseConfiguration.getConfiguration().getProperties();
 
-            logger.debug("Initializing maps");
-            eligibleJobsDescriptorsLicenses = new HashMap<String, LinkedBlockingQueue<JobDescriptorImpl>>(properties.size());
-            eligibleTasksDescriptorsLicenses = new HashMap<String, LinkedBlockingQueue<EligibleTaskDescriptor>>(properties.size());
-            remainingTokens = new HashMap<String, Integer>(properties.size());
+            logger.debug("Initializing LicenseSynchronization");
+            licenseSynchronization = new LicenseSynchronization();
 
             Enumeration propertyNames = properties.propertyNames();
             while (propertyNames.hasMoreElements()) {
                 String software = (String) propertyNames.nextElement();
-                int maxNbTokens = Integer.parseInt(properties.getProperty(software));
-
-                if (maxNbTokens > 0) {
+                int nbTokens = Integer.parseInt(properties.getProperty(software));
```

Comment:
please catch eventual exception and add a logger error message in case of wrong format

Reply:
parseInt already throws NumberFormatException

Reply:
Ok.